elastic/kibana
# [Discover] Improve warn feedback when data view has changed after rule creation

## Summary

Closes #134232

This PR reduces cases Data view changed toast will appearance. It appears only on changing one of the following parameters.
https://github.com/elastic/kibana/blob/317db097eead455809242802492d50f6f195a12d/src/plugins/discover/public/application/view_alert/view_alert_route.tsx#L23



### Test notes
- Create an output index and data view `test`
```
PUT test
{
    "settings" : {
        "number_of_shards" : 1
    },
    "mappings" : {
        "properties" : {
            "rule_id" : { "type" : "text" },
            "rule_name" : { "type" : "text" },
            "alert_id" : { "type" : "text" },
            "context_message": { "type" : "text" }
        }
    }
}
```
- Create alerts connector using `test` index
- Open `esQuery` alert in `KQL or Lucene` mode or just using Discover `Alerts` button.
- Specify the following params: `IS ABOVE: 1`, `FOR THE LAST: 30 min`.
- Try execute it by clicking `Test query`. It should match some results.
- When choosing connector, use the following config
```
{
    "rule_id": "{{rule.id}}",
    "rule_name": "{{rule.name}}",
    "alert_id": "{{alert.id}}",
    "context_message": "{{context.message}}"
}
```
- Create the alert
- On management, change one of data view settings which might affect the results (see full list above). For example add new source field filter.
- Find the alert document by browsing `test` data view in Discover
- Follow the link from the document triggered by alert  
- See warning toast which notifies about `DataView` change.

### Checklist
